docs/docs/migration-guide.mdx

```diff
@@ -10,6 +10,37 @@ description: |
 This page contains information about changes between major versions and
 how you can migrate from one version to another.
 
+## Rasa 2.3.3 to Rasa 2.3.4
+
+:::caution
+This is a release **breaking backwards compatibility of machine learning models**.
+It is not possible to load previously trained models if they were trained with `model_confidence=cosine` or
+`model_confidence=inner` setting. Please make sure to retrain the assistant before trying to use it with this improved version.
+
+:::
+
+### Machine Learning Components
+
+Rasa Open Source `2.3.0` introduced the option of using cosine similarities for model confidences by setting
+`model_confidence=cosine`. Some post-release experiments revealed that doing so can change the order of predicted
+labels. Since this is wrong, the option `model_confidence=cosine` has been removed and is no longer available.
+
+Also, `model_confidence=inner` is deprecated as it produces an unbounded range of confidences which can break
+the logic of assistants in various other places. Users are now encouraged to try `model_confidence=linear_norm`
+which will produce a linearly normalized version of dot product similarities with
+each value in the range `[0,1]`. Setting `model_confidence` to `linear_norm` will be helpful when trying to tune
+fallback thresholds.
+
+If the assistant was trained with `model_confidence=cosine` or `model_confidence=inner` setting, it should be
+re-trained by either removing the `model_confidence` option from the configuration or setting it to `linear_norm`, as:
+```
+- name: DIETClassifier
+  model_confidence: linear_norm
+  constrain_similarities: True
+  ...
```

Comment:
```suggestion
the logic of assistants in various other places. 

We encourage you to try `model_confidence=linear_norm` which will produce a linearly normalized version of dot product similarities with each value in the range `[0,1]`. This can be done with the following config:
```
- name: DIETClassifier
   model_confidence: linear_norm
   constrain_similarities: True
```

If you trained a model with `model_confidence=cosine` or `model_confidence=inner` setting using previous versions of Rasa Open Source, please re-train by either removing the `model_confidence` option from the configuration or setting it to `linear_norm`.
```